R/SEIaImIsRD.R

```diff
@@ -0,0 +1,356 @@
+#' @export
+setMethod("initial_conditions", "SEIaImIsRD",
+          function(object) object@initial_conditions)
+
+
+#' @describeIn SEIaImIsRD Retrieves transmission parameters of SEIaImIsRD model.
+#'
+#' @param object An object of the class SEIaImIsRD.
+#' @aliases initial_conditions,SEIaImIsRD-method
+#' @export
+setMethod("transmission_parameters", "SEIaImIsRD",
+          function(object) object@transmission_parameters)
+
+
+
+#' @describeIn SEIaImIsRD Setter method for initial population sizes (in fraction)
+#' of the SEIaImIsRD model.
+#'
+#' @param object An object of class SEIaImIsRD
+#' @param value A numeric named list containing the initial fraction of population groups:
+#' S - susceptible,
+#' E - exposed,
+#' I_asymptomatic - infected with no symptom,
+#' I_mild - infected with mild symptoms,
+#' I_severe - infected severe symptoms that need further hospitalization,
+#' D - dead due to the infection,
+#' R - recovered.
+#'
+#' @return An object of class SEIaImIsRD with initial population and parameters
+#' @aliases initial_conditions<-,SEIaImIsRD
+#' @export
+#'
+setMethod(
+  "initial_conditions<-",
+  "SEIaImIsRD",
+  function(object,
+           value = list(S = NA_real_, E = NA_real_,
+           I_asymptomatic = NA_real_, I_mild = NA_real_, I_severe = NA_real_,
+           R = NA_real_,
+           D = NA_real_)) {
+    init_pop_list <- value
+    names(init_pop_list) <- object@initial_condition_names
+    object@initial_conditions <- init_pop_list
+    # check if values are valid
+    errors <- character()
+    # check whether all required initial population groups are set
+    is_na_pop <- is.na(init_pop_list)
+    if (sum(is_na_pop) != 0) {
+      msg <- paste("Missing initial setting for population group:",
+                   paste(names(init_pop_list)[is_na_pop], collapse = ", ")
+      )
+      errors <- c(errors, msg)
+    }else{
+      # check whether the sum of initial population is normalized to 1
+      sum_init_pop <- sum(unlist(init_pop_list))
+      if (sum_init_pop != 1) {
+        msg <- "Sum of initial population is not 1, please normalize"
+        errors <- c(errors, msg)
+      }
+    }
+    if (length(errors) == 0) {
+      object@initial_conditions <- init_pop_list
+      object
+      }
+    else stop(paste(errors, ", please check and rerun transmission_parameters<-.\n"))
+  })
+
+
+#' @describeIn SEIaImIsRD Setter method for transmission parameters
+#' of the SEIaImIsRD model.
+#'
+#' @param object An object of class SEIaImIsRD
+#' @param value A numeric named list of values for transmission parameters:
+#' beta - a list of the effective contact rate from each infected group (i.e. rate at which an infected individual exposes susceptible),
+#' kappa - rate of progression from exposed to infectious (the reciprocal is the incubation period),
+#' omega - rate at which recovered individuals become susceptible,
+#' p_symptom - a list of fraction of different infected groups,
```

Comment:
What's the order of this list? Is it worth making this a named list?

Reply:
Also, does it sum to 1? It may be worth stating this explicitly.

Reply:
In p_symptom users only need to specify mild and severe groups. And the sum of the two should be <=1. I've included that in checks and docstrings.